### ISSUE TITLE
How are deactivated structures handled when a PW battle starts?

### ISSUE THREAD

**Issue by GoogleFrog** [ZeroK-RTS/Zero-K-Infrastructure#2138]
Ideally deactivated PW structures would be sent ingame but with the key that indicates that they are deactivated. Any deactivated structure which is destroyed in a PW battle would be re-deactivated.

I am not necessarily saying that a change is required here, but I would like to know how it works.

**Comment by DeinFreund** [comment]
There used to be a `isDestroyed` key. Currently it is not sent to the game whether the structure is active or inactive and all structures are sent. It should be easy to send a true/false whether the structure is currently active. How should the key be called?

**Comment by Anarchid** [comment]
Deactivated could also mean freshly built and still powering up. Maybe `Inactive` instead.

**Comment by GoogleFrog** [comment]
isInactive sounds like a good key. Given the current implementation, I would intend for the key to disable the abilities of the PW structure ingame.

Now answer this question: If a planet has an active wormhole and a deactivated improved wormhole then what is sent ingame? Ideally both would be sent ingame so you can destroy the wormhole.

Currently, an improved wormhole prevents wormhole being sent ingame. I may remove this ability from the DB just to make things simpler.

**Comment by DeinFreund** [comment]
I think all structures with `IngameUnitName` are sent to the game? How is the adv wormhole deduplication implemented?

**Comment by GoogleFrog** [comment]
Wormhole deduplication was game-side. I just removed it because it is not worth the complications that arise. I came up with a rule to handle evacuations with duplicate wormholes:
 - All wormholes are spawned.
 - The most powerful wormhole is the one that is used. Slow damage, emp etc.. on this wormhole all impact the speed at which evac charges.
 - If the most powerful wormhole is destroyed then all teleportation is cancelled and the second most powerful wormhole is used.
 - Charge is not lost on destruction. So destroying an Improved wormhole at 4 out of 5 minutes charge results in a basic wormhole at 4 out of 8 minutes charge.
 - Destroying a lesser wormhole while a better one is active has no effect on current evacuations.
